### PR TITLE
Cluster repair tool

### LINKED ISSUE #1789: Recover a cluster with controller.web scaled to 0
If you have no controller web instances, it's hard to start one up again with 'flynn scale' commands.

### LINKED ISSUE #1893: vagrant: Single machine multi-node test cluster doesn't work on Vagrant VM
After successful bootstrap fails with:

```
Error writing CA certificate: Get https://controller.dev.localflynn.com/ca-cert: dial tcp 192.0.2.100:443: connection refused
```

New DNS records are probably required to support this behaviour.

### LINKED ISSUE #1988: Cluster fails to resurrect after non-graceful shutdown
I setup a 3 node cluster, then powered off the VMs running the cluster. After booting up again the cluster failed to start. Debug output:

* https://gist.github.com/cab0a5330b81681217bc
* https://gist.github.com/507775cdb47049aacb54
* https://gist.github.com/62f781b53660e0a892bc

### LINKED ISSUE #2046: Single node resurrection
Hi,

I am playing with flynn in order to better understand it.
So after a few days of testing I decided to reboot the node (I have a single node setup) and see if I am able to resurrect it.

After reboot flynn-host did not start on its own and I had to start it manually.
Thing is that `flynn-host ps` returns: ..127.0.0.1:1111: connection refused.. (I can provide debug gist url if needed)

According to some issues I read, this feature should work. The node should come up on its own without human intervention.
I can't find any sort of documentation in regards to cluster restoration or maybe a recovery procedure.

Is there something documented for this scenario?

And let's assume I had 3 nodes setup but all 3 went down at the same time. Will the cluster be able to recover?

### PR DESCRIPTION
**Pull request by titanous**
Adds `flynn-host fix` which repairs a variety of common cluster ailments.

Closes #2046
Closes #1789
Closes #1988
Closes #1893

### CODE REVIEW

**Comment by josephglanville** [comment]
This looks awesome. Also fixes #1893 if I am not mistaken.

File: host/cli/fix.go [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+		return fmt.Errorf("invalid or missing --min-hosts value")
+	}
+
+	f.hosts, err = c.Hosts()
+	if err != nil {
+		f.l.Error("unable to list hosts from discoverd, falling back to peer IP list", "error", err)
+		var ips []string
+		if ipList := args.String["--peer-ips"]; ipList != "" {
+			ips = strings.Split(ipList, ",")
+			if minHosts == 0 {
```
Comment (lmars) [inline host/cli/fix.go]: `minHosts` can't be zero here due to the checks above

File: host/cli/fix.go [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+				minHosts = len(ips)
+			}
+		}
+		if len(ips) == 0 {
+			return fmt.Errorf("error connecting to discoverd, use --peer-ips: %s", err)
```
Comment (lmars) [inline host/cli/fix.go]: Drop the `err` argument

Comment (lmars) [inline host/cli/fix.go]: Oh unless you want it there, didn't realise this was in an `if err != nil` block

File: host/cli/fix.go [hunk 3]
```diff
@@ -0,0 +1,607 @@
...
+	}
+	f.l.Info("waiting for job to start")
+	return discoverd.GetInstances(service, time.Minute)
+}
+
+func (f *clusterFixer) fixDiscoverd() error {
+	f.l.Info("ensuring discoverd is running on all hosts")
+	releases := f.findAppReleaseJobs("discoverd", "app")
+	if len(releases) == 0 {
+		return fmt.Errorf("didn't find any discoverd release jobs")
```
Comment (lmars) [inline host/cli/fix.go]: This returns an error on a freshly booted cluster as there are no discoverd jobs in either the [done or crashed state](https://github.com/flynn/flynn/blob/first-aid/host/cli/fix.go#L563-L565) (I was trying to use `flynn-host fix` to resurrect a scheduler that I killed)

**Comment by lmars** [comment]
@titanous I pushed a fix, LGTM

**Pull request merged**